### PR TITLE
Collection Cluster API via GRPC

### PR DESCRIPTION
**Pull request by nilhex**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally using `cargo fmt` command prior to submission?
3. [x] Have you checked your code using `cargo clippy` command?

GRPC implementation of Collection Cluster API. 
As requested by https://github.com/qdrant/rust-client/issues/32
Related with https://github.com/qdrant/rust-client/pull/33

### CODE REVIEW

**Comment by agourlay** [comment]
Thanks for your contribution!

Please target the `dev` branch for this PR.

Also the Clippy error is unrelated to your change and has already been fixed on `dev`.

**Comment by nilhex** [comment]
@agourlay Got your point. It's done.

File: src/tonic/api/collections_api.rs [hunk 1]
```diff
@@ -166,6 +170,102 @@ impl Collections for CollectionsService {
...
+        &self,
+        request: Request<CollectionClusterInfoRequest>,
+    ) -> Result<Response<CollectionClusterInfoResponse>, Status> {
+        validate(request.get_ref())?;
+        let cluster_info = do_get_collection_cluster(
+            self.dispatcher.toc(),
+            request.into_inner().collection_name.as_str(),
+        )
+        .await
+        .map_err(|e| Status::new(tonic::Code::Unknown, e.to_string()))?;
```
Comment (agourlay) [inline src/tonic/api/collections_api.rs:184]:
```suggestion
        .map_err(error_to_status)?;
```

File: src/tonic/api/collections_api.rs [hunk 2]
```diff
@@ -166,6 +170,102 @@ impl Collections for CollectionsService {
...
+                            peer_id: op.peer_id,
+                        },
+                    })
+                }
+            },
+            self.dispatcher.as_ref(),
+            timeout.map(std::time::Duration::from_secs),
+        )
+        .await
+        .map_err(|e| Status::new(tonic::Code::Unknown, e.to_string()))?;
```
Comment (agourlay) [inline src/tonic/api/collections_api.rs:264]:
```suggestion
        .map_err(error_to_status)?;
```

File: src/tonic/api/collections_api.rs [hunk 3]
```diff
@@ -166,6 +170,102 @@ impl Collections for CollectionsService {
...
+
+        let response = CollectionClusterInfoResponse {
+            peer_id: cluster_info.peer_id,
+            shard_count: cluster_info.shard_count as u64,
+            local_shards: cluster_info
+                .local_shards
+                .iter()
+                .map(|shard| LocalShardInfo {
```
Comment (agourlay) [inline src/tonic/api/collections_api.rs:192]: we usually place the `gRPC` object conversions [here](https://github.com/qdrant/qdrant/blob/master/lib/api/src/grpc/conversions.rs)

**Review by agourlay — COMMENTED** [review]
Did a first review pass, it looks promising :+1:

**Comment by nilhex** [comment]
@agourlay I have corrected your comments. Please check if everything is ok with placing conversions.

File: src/tonic/api/collections_api.rs (hunk 3 above)
Comment (agourlay) [reply src/tonic/api/collections_api.rs]: Good catch :+1: 

Sorry I picked the wrong `conversions.rs` file, happy you were able to figure this on your own.

**Review by agourlay — APPROVED** [review]
LGTM :+1:

**Review by generall — APPROVED** [review]

**Pull request merged**